daml-foundations/daml-ghc/ghc-compiler/src/DA/Daml/GHC/Compiler/UtilLF.hs

```diff
@@ -167,5 +159,22 @@ ghcTypes = Module
           (ERecCon (TypeConApp (qual (dataTypeCon dataProxy)) [TVar (mkTypeVar "a")]) [])
       }
 
+mkVariantOrEnumDataCons :: Version -> [VariantConName] -> DataCons
+mkVariantOrEnumDataCons version cons
```

Comment:
I'd just call this mkDataEnum and treat it as a smart contstructor that knows about the fallback.

Reply:
Make sense.